modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/jdbc/JdbcThinFeature.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.odbc.jdbc;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import org.apache.ignite.internal.ThinProtocolFeature;
+
+/**
+ * Defines supported features for JDBC thin client.
+ */
+public enum JdbcThinFeature implements ThinProtocolFeature {
+    RESERVED(0);
+
+    /** */
+    private static final EnumSet<JdbcThinFeature> ALL_FEATURES_AS_ENUM_SET = EnumSet.allOf(JdbcThinFeature.class);
+
+    /** */
+    private static final byte[] ALL_FEATURES_AS_BYTES = ThinProtocolFeature.featuresAsBytes(JdbcThinFeature.values());
+
+    /** Feature id. */
+    private final int featureId;
+
+    /**
+     * @param id Feature ID.
+     */
+    JdbcThinFeature(int id) {
+        featureId = id;
+    }
+
+    /** {@inheritDoc} */
+    @Override public int featureId() {
+        return featureId;
+    }
+
+    /**
+     * @param bytes Feature byte array.
+     * @return Set of supported features.
+     */
+    public static EnumSet<JdbcThinFeature> enumSet(byte[] bytes) {
+        return ThinProtocolFeature.enumSet(bytes, JdbcThinFeature.class);
+    }
+
+    /**
+     * @return Byte array representing all supported features by current node.
+     */
+    public static byte[] allFeaturesAsBytes() {
```

Comment:
I don't understand, why does `JdbcThinFeature` provides custom `allFeaturesAsBytes()` when `ThinProtocolFeature` has `featuresAsBytes` that seems to do the same as method description suggests. What's the purpose (e.g. usecase) of `featuresAsBytes` then?

Reply:
Take a look at the implementation.
`ThinProtocolFeature#featuresAsBytes(E[])` based on the loop.
`JdbcThinFeature#allFeaturesAsBytes` use cached value.